### ISSUE TITLE
CUDA Convnet keeps failing

### ISSUE THREAD

**Issue by salmanulhaq** [liuliu/ccv#85]
I'm trying to train a new image dataset. The train and test files conform with the required format but when I try to train, it fails to correctly assert the GPU pointers being allocated. I manually checked it and the GPU mem pointers do get allocated but assertion fails. I removed all assertions and then it fails to allocate device pointers within one of the structures.

I woud love to have more documentation on training a custom image dataset.

**Comment by liuliu** [comment]
How much GPU memory do you have? Most cases these are you don't have enough GPU memory for allocation. Try:
1). Using a smaller mini_batch size;
2). Get a bigger GPU memory;
with mini batch size of 256 and imageNet network topology, it will allocate 4GB+ memory.

**Comment by salmanulhaq** [comment]
Thanks for the quick reply. Actually it was a GPU memory issue. I discounted the mem allocations in iteration :)

I'm trying to train a very small model (2 classes 6 images each class) with the same network topology as in the image-net example on a Ubuntu x64 12.04 box. I'm getting a very weird error within GSL during random number generation ( gsl_ran_shuffle() ) and gives me "n is either 0 or greater than the largest possible value.

**Comment by liuliu** [comment]
Not sure about the gsl problem.

In your case, you have too little examples, consider using the pretrained data, and chop off the last layer and train that output (a 4096 dim vector) with SVM.

This version doesn't make this particular procedure easy, you need to dig into the code a bit.

The way to do this is documented in: http://arxiv.org/pdf/1403.6382v1.pdf

**Comment by salmanulhaq** [comment]
Is there a lower limit on the training data? Also, have you tried training on a Ubuntu 12.04 x64 machine? Would be great to have an example on training a custom image set. Still digging into it and would appreciate pointers :)

**Comment by liuliu** [comment]
I think for this size of network, to train from scratch, a million of examples is probably the lowest bound. I am working on an interface to standardize the training with a few examples and the pre-trained model.
